src/v/cluster/partition.h

```diff
@@ -137,6 +137,8 @@ class partition {
         return _raft->linearizable_barrier();
```

Comment:
```
compaction: decompress batch in kafka layer
```
can you describe the strategy for how data is moving across cores? which core incurs the cost of decompression, etc?

Reply:
We dispatch a produce request to the right core pretty early in the processing pipeline, see `produce_topic_partition` function of `produce.cc`. It allows us to move decompression from the storage layer (segment.cc) to the kafka layer without crossing the core boundary so it still happens on the same core as it used before.